packages/sort_box/lib/src/sort_box.dart

```diff
@@ -51,6 +54,32 @@ class SortBox<T extends Enum, R> {
     input.sort((item1, item2) => _compare(item1, item2, boxes.iterator..moveNext()));
   }
 
+  /// Sorts the [input] list according to their [box].
+  ///
+  /// A box contains the property and [SortBoxOrder] how the list should be sorted.
+  /// In case the property of two elements is considered equal all following boxes specified at `boxes[property]` are applied.
+  /// If specified [presort] will be applied before [box] and [boxes].
+  ///
+  /// See:
+  ///   * [sort] to sort a `List`
+  void sortBuilder(
+    ListBuilder<R> input,
+    Box<T> box, [
+    Set<Box<T>>? presort,
+  ]) {
+    if (input.length <= 1) {
+      return;
+    }
+
+    final boxes = {
+      ...?presort,
+      box,
+      ...?this.boxes[box.property],
+    };
+
+    input.sort((item1, item2) => _compare(item1, item2, boxes.iterator..moveNext()));
```

Comment:
I don't like that this is completely duplicate with the existing method (but I know it won't work because the sort methods are not the same. How about calling a common method and give it `input.sort` as a callback?

Reply:
I think this would only make it more complex and ugly.
With the entire package being only 100 LOC we should consider leaving this duplication for now.

Reply:
Ok